lib/core/test/unit/Cardano/Wallet/Primitive/AddressDerivation/MintBurnSpec.hs

```diff
@@ -0,0 +1,237 @@
+import Cardano.Wallet.Primitive.AddressDerivation
+    ( Depth (..)
+    , DerivationType (..)
+    , Index (..)
+    , Passphrase
+    , WalletKey (publicKey)
+    , getRawKey
+    , hashVerificationKey
+    , liftRawKey
+    )
+import Cardano.Wallet.Primitive.AddressDerivation.MintBurn
+    ( derivePolicyKeyAndHash, derivePolicyPrivateKey )
+import Cardano.Wallet.Primitive.AddressDerivation.Shelley
+    ( ShelleyKey )
+import Cardano.Wallet.Primitive.AddressDerivationSpec
+    ()
+import Cardano.Wallet.Unsafe
+    ( unsafeBech32Decode, unsafeFromHex, unsafeMkMnemonic, unsafeXPrv )
+import Data.Function
+    ( (&) )
+import Data.Text
+    ( Text )
+import GHC.TypeNats
+    ( KnownNat )
+import Test.Hspec
+    ( Expectation, Spec, describe, it, shouldBe )
+import Test.Hspec.Extra
+    ( parallel )
+import Test.QuickCheck
+    ( Arbitrary (..), Property, property, vector, (=/=), (===) )
+import Test.QuickCheck.Arbitrary
+    ( arbitraryBoundedEnum )
+
+import qualified Cardano.Address.Script as CA
+import qualified Cardano.Wallet.Primitive.AddressDerivation.Shelley as Shelley
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy as BL
+import qualified Data.Text.Encoding as T
+
+spec :: Spec
+spec = do
+    parallel $ describe "Mint/Burn Policy key Address Derivation Properties" $ do
+        it "Policy key derivation from master key works for various indexes" $
+            property prop_keyDerivationFromXPrv
+        it "Policy public key hash matches private key" $
+            property prop_keyHashMatchesXPrv
+        it "The same index always returns the same private key" $
+            property prop_keyDerivationSameIndexSameKey
+        it "A different index always returns a different private key" $
+            property prop_keyDerivationDiffIndexDiffKey
+        it "Using derivePolicyKeyAndHash returns same private key as using derivePolicyPrivateKey" $
+            property prop_keyDerivationRelation
+        it "Deriving a policy key with cardano-address returns same result as cardano-wallet" $ do
+            unit_comparePolicyKeys goldenTestMnemonic (Index 0x80000000)
+                "acct_xsk1tqqqnvtppk994fxm05wtlvp6uuu58srue9st8ew29vr5mxxf89tj2ze05pm8qjkyfetpzl58jkgx6dd3s96szhyxfajpc2gxv22xef0ems2cnvh5d5xjemgghg9m8489v6c9rvnt9za2pruyrtkp5y77l5ujxeug"
+                -- Child key 1855H/1815H/0H generated by cardano-address CLI
+                -- from test mnemonic
+            unit_comparePolicyKeyHashes goldenTestMnemonic (Index 0x80000000)
+                "6adb501348b99cd38172f355615d7c1d0b9d1e3fc69b565b85a98127"
+                -- Hash of child key 1855H/1815H/0H generated by cardano-address
+                -- CLI from test mnemonic
+            unit_comparePolicyKeys goldenTestMnemonic (Index 0x80000010)
+                "acct_xsk1rprds8krzhspy8xhkupk2270hzkexmkdga98zv5ns4x9d9kf89t4qp438a26sn62wsfldthkrvnwe6vnetuehgxyxt4hm46zsw03mknuwef0afcljmultpjjwx8zqm0ftnfkqucwvf5qr4gved6h3gnyeqcsur4p"
```

Comment:
cardano-address did not produce the correct bech32 prefix for policy keys, but never mind.